scripts/frida/frida_api_models.py

```diff
@@ -32,30 +32,43 @@ def validate_java_api_field_consistency(self):
         return self
 
 
-class JniApi(BaseModel):
-    pass
-
+SUPPORTED_NATIVE_TYPES = {
+    'int', 'uint', 'bool',
+    'char*', 'const char*'
+}
+UNSUPPORTED_NATIVE_TYPES = {
+    'long', 'ulong', 'size_t', 'float', 'double',
+    'void*', 'pointer'
+}
+VALID_NATIVE_TYPES = SUPPORTED_NATIVE_TYPES.union(UNSUPPORTED_NATIVE_TYPES)
 
 class NativeApi(BaseModel):
-    pass
-
+    library: str
+    function: str
+    arguments: bool = True
+    argument_types: List[str] = Field(default_factory=list)
 
-class JavaSection(BaseModel):
-    methods: List[JavaApi] = Field(default_factory=list)
+    @model_validator(mode='after')
+    def validate_argument_types(self):
+        print(f'Reminder: supported types are {SUPPORTED_NATIVE_TYPES}. '
```

Comment:
Let's avoid printing messages from the model module. Instead, raise an exception with a custom message and let the caller decide what to do with it.